code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm

```diff
@@ -7,6 +7,26 @@
 	unacidable = 1 //glass
 	center_of_mass = list("x"=16, "y"=10)
 
+/obj/item/weapon/reagent_containers/food/drinks/standard_feed_mob(var/mob/user, var/mob/target)
+	// Special snowflake code inbound.
+	if(is_open_container() && user == target && user.a_intent != I_HELP && user.a_intent != I_HURT) // This is a pretty specific case.
```

Comment:
this is not really a specific case, and will be frequently done by accident. 

Make it a rightclick verb instead please